google-cloud-storage/src/test/java/com/google/cloud/storage/conformance/retry/RpcMethod.java

```diff
@@ -0,0 +1,167 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.storage.conformance.retry;
+
+import java.util.Arrays;
+import java.util.Map;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+interface RpcMethod {
+
+  String getFullyQualifiedMethodName();
+
+  /**
+   * Enumerate the hierarchy of storage rpc methods.
+   *
+   * <p>These class names intentionally do not follow java convention, because they are mapping
+   * directly to lower level values.
+   */
+  final class storage {
+    private static String getFullQualifiedMethodName(Enum<?> e) {
+      return String.format("storage.%s.%s", e.getClass().getSimpleName(), e.name());
+    }
+
+    enum bucket_acl implements RpcMethod {
+      delete,
+      get,
+      insert,
+      list,
+      patch,
+      update;
+
+      @Override
+      public String getFullyQualifiedMethodName() {
+        return getFullQualifiedMethodName(this);
+      }
+    }
+
+    enum buckets implements RpcMethod {
+      delete,
+      get,
+      insert,
+      list,
+      patch,
+      update,
+      getIamPolicy,
+      lockRetentionPolicy,
+      setIamPolicy,
+      testIamPermissions;
+
+      @Override
+      public String getFullyQualifiedMethodName() {
+        return getFullQualifiedMethodName(this);
+      }
+    }
+
+    enum default_object_acl implements RpcMethod {
+      delete,
+      get,
+      insert,
+      list,
+      patch,
+      update;
+
+      @Override
+      public String getFullyQualifiedMethodName() {
+        return getFullQualifiedMethodName(this);
+      }
+    }
+
+    enum hmacKey implements RpcMethod {
+      delete,
+      get,
+      list,
+      update,
+      create;
+
+      @Override
+      public String getFullyQualifiedMethodName() {
+        return getFullQualifiedMethodName(this);
+      }
+    }
+
+    enum notifications implements RpcMethod {
+      delete,
+      get,
+      insert,
+      list;
+
+      @Override
+      public String getFullyQualifiedMethodName() {
+        return getFullQualifiedMethodName(this);
+      }
+    }
+
+    enum object_acl implements RpcMethod {
+      delete,
+      get,
+      insert,
+      list,
+      patch,
+      update;
+
+      @Override
+      public String getFullyQualifiedMethodName() {
+        return getFullQualifiedMethodName(this);
+      }
+    }
+
+    enum objects implements RpcMethod {
+      delete,
+      get,
+      insert,
+      list,
+      patch,
+      update,
+      compose,
+      rewrite,
+      copy;
+
+      @Override
+      public String getFullyQualifiedMethodName() {
+        return getFullQualifiedMethodName(this);
+      }
+    }
+
+    enum serviceaccount implements RpcMethod {
+      get,
+      put;
```

Comment:
ServiceAccount API doesn't have a PUT operation, is this intended?

Reply:
I've cleaned this up.